PickMeUp/PickMeUp/Model/PMDataManager.m

```diff
@@ -0,0 +1,71 @@
+//
+//  PMDataManager.m
+//  PickMeUp
+//
+//  Created by Abigail Shilts on 7/27/22.
+//
+#import "Parse/Parse.h"
+#import "PMCachingFunctions.h"
+#import "PMDataManager.h"
+#import "PMTree.h"
+#import "StringsList.h"
+
+@interface PMDataManager ()
+@property (nonatomic, copy) void (^block)(NSArray<PMConversation *> *);
+@end
+
+@implementation PMDataManager
+
++ (id)dataManager {
+    static PMDataManager *dataManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        dataManager = [[self alloc] init];
+    });
+    return dataManager;
+}
+
+-(void)fillConversations:(void(^)(NSArray<PMConversation *> *))block {
+    if (block == nil) {
```

Comment:
You should still do all the work, but check `block` against nil just before you call it. Imagine a case when you just want to update the dataManager cache.

Let's rename `block` to `completion` or `completionBlock`

Also, in the header file you should declare `block` as nullable